csrc/megatron/fused_weight_gradient_dense_cuda.cu

```diff
@@ -10,10 +10,481 @@
+                                                          matC,
+                                                          matD,
+                                                          pref,
+                                                          request_solutions,
+                                                          heuristicResult,
+                                                          &returnedAlgoCount));
+
+    if(returnedAlgoCount == 0)
+    {
+        std::cerr << "No valid solution found!" << std::endl;
+        return;
+    }
+
+    uint64_t workspace_size = 0;
+    for(int i = 0; i < returnedAlgoCount; i++)
+        workspace_size = max(workspace_size, heuristicResult[i].workspaceSize);
+    // In this sample, the workspace is already allocated with max_workspace_size
+    // If not, allocate d_workspace here
+    // CHECK_HIP_ERRORhipMalloc(&d_workspace, workspace_size));
+
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmul(handle,
+                                          matmul,
+                                          &alpha,
+                                          A,
+                                          matA,
+                                          B,
+                                          matB,
+                                          &beta,
+                                          C,
+                                          matC,
+                                          D,
+                                          matD,
+                                          &heuristicResult[0].algo,
+                                          d_workspace,
+                                          workspace_size,
+                                          stream));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutDestroy(matA));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutDestroy(matB));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutDestroy(matC));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutDestroy(matD));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulDescDestroy(matmul));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulPreferenceDestroy(pref));
+    return;
+}
+
+
+// FP32 wrapper around cublas GEMMEx
+void gemmex_wrapper(
+    hipblasLtHandle_t handle,
+    hipblasOperation_t transa,
+    hipblasOperation_t transb,
+    int m,
+    int n,
+    int k,
+    int batch_count,
+    float& alpha,
+    float& beta,
+    float* A,
+    float* B,
+    float* C,
+    float* D,
+    void*   d_workspace,
+    int64_t  max_workspace_size,
+    hipStream_t   stream) {
+    hipblasLtMatrixLayout_t matA, matB, matC, matD;
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutCreate(&matA, HIP_R_32F, m, k, m));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutCreate(&matB, HIP_R_32F, n, k, n));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutCreate(&matC, HIP_R_32F, m, n, m));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatrixLayoutCreate(&matD, HIP_R_32F, m, n, m));
+
+    if(batch_count > 1)
```

Comment:
As we already have folded the batch dimension at higher level calls, we don't need to deal with `batch_count>1` in the `gemmex_wrapper` and only need to deal with `batch_count=1` for simplicity.